toolchain/lowering/lowering_handle.cpp

```diff
@@ -65,11 +72,11 @@ auto LoweringHandleFunctionDeclaration(LoweringContext& context,
   llvm::SmallVector<llvm::Type*> args;
   args.resize_for_overwrite(param_refs.size());
   for (int i = 0; i < static_cast<int>(param_refs.size()); ++i) {
-    args[i] = context.LowerNodeToType(
+    args[i] = context.GetLoweredNodeAsType(
         context.semantics_ir().GetNode(param_refs[i]).type_id());
   }
 
-  llvm::Type* return_type = context.LowerNodeToType(
+  llvm::Type* return_type = context.GetLoweredNodeAsType(
```

Comment:
Assuming you need to not use the `void` LLVM type for empty tuples, I'd add code here to use that for the return type when no return type is specified (and we expect no expression in the return statement). That should also remove some of the churn in the tests below.

Reply:
Added a TODO to investigate special-casing the type as `void` everywhere, since in theory that could be done (e.g., we could elide variables that would otherwise have a type of void). But since this would be a pretty sweeping change, I'm leaving it as a TODO.

I think there may have been some confusion too about whether `fn F()` and `fn F() -> ()` were intended to be equivalent. That comes from https://github.com/carbon-language/carbon-lang/issues/443 and was the accepted design in https://github.com/carbon-language/carbon-lang/pull/438. There is no "void" or "lack of return" there.